warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/IngestJob.java

```diff
@@ -455,9 +455,14 @@ public int run(String[] args) throws Exception {
                     numRecords += recordC.getValue();
                 }
                 // records that throw runtime exceptions are still counted as processed
-                float percentError = 100 * ((float) numExceptions / numRecords);
-                log.info(String.format("Percent Error: %.2f", percentError));
-                if (conf.getInt("job.percent.error.threshold", 101) <= percentError) {
+                if (numRecords > 0) {
+                    float percentError = 100 * ((float) numExceptions / numRecords);
+                    log.info(String.format("Percent Error: %.2f", percentError));
+                    if (conf.getInt("job.percent.error.threshold", 101) <= percentError) {
+                        return jobFailed(job, runningJob, outputFs, workDirPath);
+                    }
+                } else {
+                    log.info("Job ran with no records");
```

Comment:
Is it really a failure if there are no records?  I would think if you saw 0 records, then your percent error would 0%.

```suggestion
                float percentError = (numRecords == 0) ? 0.0 : 100 * ((float) numExceptions / numRecords);
                log.info(String.format("Percent Error: %.2f", percentError));
                if (conf.getInt("job.percent.error.threshold", 101) <= percentError) {
```

Reply:
SonarQube was trying to avoid the divide by zero, I like your solution.